cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -100,6 +99,44 @@ def set_app_version():
     hookenv.application_version_set(version.split(b' v')[-1].rstrip())
 
 
+@when('kubernetes.worker.bins.installed')
+@when_not('kube-dns.available')
+def notify_user_transient_status():
+    ''' Notify to the user we are in a transient state and the application
+    is still converging. Potentially remotely, or we may be in a detached loop
+    wait state '''
+
+    # During deployment the worker has to start kubelet without cluster dns
+    # configured. If this is the first unit online in a service pool waiting
+    # to self host the dns pod, and configure itself to query the dns service
+    # declared in the kube-system namespace
+
+    hookenv.status_set('waiting',
+                       'Waiting for cluster-manager to initiate start')
+
+
+@when('kubernetes.worker.bins.installed', 'kube-dns.available')
+def notify_user_converging_status(kube_dns):
+    ''' There are different states that the worker can be in, where we are
+    waiting for dns, waiting for cluster turnup, or ready to serve
+    applications'''
+    # Daemon options are managed by the FlagManager class
+    kubelet_opts = FlagManager('kubelet')
+
+    # Query the FlagManager dict for the dns option, and determine if
+    # kubelet is running
+    if (_systemctl_is_active('kubelet') and
+       '--cluster-dns' not in kubelet_opts.data):
+        hookenv.status_set('waiting', 'Waiting on cluster dns')
+    elif(_systemctl_is_active('kubelet') and
+         '--cluster-dns' in kubelet_opts.data):
+        hookenv.status_set('active',
+                           'Kubernetes worker running')
+    # if kubelet is not running, we're waiting on something else to converge
+    elif(not _systemctl_is_active('kubelet')):
+        hookenv.status_set('waiting', 'waiting on kubelet')
```

Comment:
We need to change this to consistent messaging. I think the other messages have Caps and full stops.